tests/integration/utils.go

```diff
@@ -1271,8 +1274,59 @@ func createDoc(
 		// Expand the slice if required, so that the document can be accessed by collection index
 		s.docIDs = append(s.docIDs, make([][]client.DocID, action.CollectionID-len(s.docIDs)+1)...)
 	}
+	startDocIndex := len(s.docIDs[action.CollectionID])
 	s.docIDs[action.CollectionID] = append(s.docIDs[action.CollectionID], docIDs...)
 
+	if action.CollectionID >= len(s.cids) {
+		// Expand the slice if required, so that the cids can be accessed by collection index
+		s.cids = append(s.cids, make([][]map[string][]cid.Cid, action.CollectionID+1)...)
+	}
+
+	endDocIndex := startDocIndex + len(docIDs) - 1
+	if endDocIndex >= len(s.cids[action.CollectionID]) {
+		s.cids[action.CollectionID] = append(
+			s.cids[action.CollectionID],
+			make([]map[string][]cid.Cid, endDocIndex-len(s.cids[action.CollectionID])+1)...,
+		)
+	}
+
+	for relativeDocIndex, docID := range docIDs {
+		results, err := nodes[0].Headstore().Query(
+			s.ctx,
+			query.Query{
+				Prefix: core.HeadStoreKey{
+					DocID: docID.String(),
+				}.ToString(),
+				KeysOnly: true,
+			},
+		)
+		if err != nil {
+			s.t.Fatalf("failed to fetch cids: %s", err.Error())
+		}
+
+		for res := range results.Next() {
+			if res.Error != nil {
+				s.t.Fatalf("failed to fetch cids: %s", res.Error.Error())
+			}
+
+			key, err := core.NewHeadStoreKey(res.Key)
+			if err != nil {
+				s.t.Fatalf("failed to fetch cids: %s", err.Error())
+			}
+
+			docIndex := startDocIndex + relativeDocIndex
+
+			if s.cids[action.CollectionID][docIndex] == nil {
+				s.cids[action.CollectionID][docIndex] = map[string][]cid.Cid{}
+			}
+
+			// This is a new doc, so we can blindly overwrite the array of cids
+			s.cids[action.CollectionID][docIndex][key.FieldID] = []cid.Cid{
+				key.Cid,
+			}
```

Comment:
thought: I assume that since this is a create function, this will never be false. You could then just do this:
```go
s.cids[action.CollectionID][docIndex] = map[string][]cid.Cid{
	key.FieldID: {key.Cid},
}
```